tests/test_cli_compile.py

```diff
@@ -619,10 +619,14 @@ def test_relative_local_url_package(pip_conf, runner, tmpdir):
     with open(in_path, "w") as reqs_in:
         reqs_in.write(f"-e file:{pkg_path_relto_reqsin.replace(os.path.sep, '/')}")
 
-    os.chdir(run_dir)
-    out = runner.invoke(
-        cli, ["-o", os.path.relpath(txt_path), os.path.relpath(in_path)]
-    )
+    start_dir = os.path.abspath(os.path.curdir)
+    try:
+        os.chdir(run_dir)
```

Comment:
It's best to move such logic into a context manager

Reply:
Actually we don't need that logic there at all! I've removed the `try`-`finally` in the test, as the test requires no more directory changing, and `pytest` will change the directory before a new test runs anyway. I've added a comment explaining that to the chdir spot in that test.

Reply:
For the remaining instance of this form, I'll close this discussion in favor of [one below](https://github.com/jazzband/pip-tools/pull/1329#pullrequestreview-611660103)